crates/bevy_app/src/app.rs

```diff
@@ -1068,4 +1103,60 @@ mod tests {
         }
         App::new().add_plugin(PluginRun);
     }
+
+    mod system_stepping {
+        use super::*;
+        use crate::app::{EventWriter, ScheduleEvent};
+        use crate::CoreSchedule;
+        use bevy_ecs::prelude::*;
+        use bevy_ecs::schedule::ExecutorKind;
+
+        #[derive(Resource, Default)]
+        struct ScheduleEvents(Vec<ScheduleEvent>);
```

Comment:
This is super weird and probably wrong: we already have `Events<ScheduleEvent>` for this.

Reply:
Ah, I think I understand.  I can just grab the `Events<ScheduleEvent>` resource directly and manually add to it without using an `EventWriter`.

I'm a little reluctant to do the testing outside of a system, as the whole purpose of the start of the tests here was to verify stepping was controllable from a system.

I did realize halfway through that the complex behavior is tested down under `Schedule`, so this test should really be "does `App` send events to the correct schedule at the end of `update()`".

I'll try making this change and see how it looks in review.

Reply:
You should be able to test that this works from within a system just fine by using `App::update` in your test?